### PR TITLE
Restore and extend ttl

### PR DESCRIPTION
**Pull request by shan-57blocks**

### CODE REVIEW

**Review by PlayJok3r — APPROVED** [review]

File: packages/sdk/src/utils/transaction.ts [hunk 1]
```diff
@@ -0,0 +1,200 @@
...
+} from './network'
+
+const handlePendingTransaction = async (
+  sendResponse: SorobanRpc.Api.SendTransactionResponse,
+  server: SorobanRpc.Server,
+) => {
+  if (sendResponse.status === 'PENDING') {
+    let getResponse = await server.getTransaction(sendResponse.hash)
+    while (getResponse.status === 'NOT_FOUND') {
+      await new Promise((resolve) => setTimeout(resolve, 1000))
```
Comment (mliu) [inline packages/sdk/src/utils/transaction.ts:27]: Should we add a maximum number of retries here? I think 1.5 minutes might be safe to assume the txn had errors.

File: packages/sdk/src/utils/transaction.ts [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+      return transactionMeta?.v3()?.sorobanMeta()?.returnValue()
+    } else {
+      throw getResponse.resultXdr
+    }
+  } else {
+    throw sendResponse.errorResult
+  }
+}
+
+export const simTransaction = async (
```
Comment (mliu) [inline packages/sdk/src/utils/transaction.ts:45]: Nit: Can we rename this to simulateTransaction?

File: packages/sdk/src/utils/transaction.ts [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+  console.log('Start extend TTL transaction')
+  const { restorePreamble } = simResponse
+  const builtTransaction = new TransactionBuilder(account, {
+    networkPassphrase: StellarNetworkPassphrase[network],
+    fee: restorePreamble.minResourceFee,
+  })
+    .setSorobanData(restorePreamble.transactionData.build())
+    .addOperation(
+      Operation.extendFootprintTtl({
+        extendTo: 3110400 - 1,
```
Comment (mliu) [inline packages/sdk/src/utils/transaction.ts:121]: Is this 6 months? My math is 178*24*60*60/5 = 3,075,840

**Review by mliu — APPROVED** [review]

File: packages/sdk/src/utils/transaction.ts (hunk 3 above)
Comment (shan-57blocks) [reply packages/sdk/src/utils/transaction.ts:121]: I use 3110400*5/24/60/60 = 180 days
And this is tested that the maximum number allowed

File: packages/sdk/src/utils/transaction.ts (hunk 1 above)
Comment (shan-57blocks) [reply packages/sdk/src/utils/transaction.ts:28]: Updated

File: packages/sdk/src/utils/transaction.ts (hunk 2 above)
Comment (shan-57blocks) [reply packages/sdk/src/utils/transaction.ts]: Updated

**Pull request merged**